src/main/resources/schema/query.graphqls

```diff
@@ -1,4 +1,17 @@
 type Query {
+  "Get a wish by ID"
   wish(id: ID): Wish
-  allWish: [Wish!]!
+
+  "Gets all Wishes"
+  allWishes: [Wish!]!
+
+  "Get all Wishes for Whiser"
+  myWishlist(userID: ID): [Wish!]!
+
+  "Get all Comments for a target(a Wish or a Comment)"
+  comments(targetID: ID): [Comment!]!
```

Comment:
If we support commenting on multiple types of items, then we need also to specify the type of commented item in query. There is no guarantee that the ID is unique for all types. In other words, there is a chance that there is a wish with id=123 and also some other type item with id=123.

Reply:
Wanted some clarification on the ID. will we use int or unique IDs?

I prefer we use unique IDs but the choice is up to you.

Reply:
The ID will be unique for a given type - there is no possibility that two wishes will have the same ID. However, if we want to enable commenting other things (other than wishes, like let's say developer teams) then there's a risk that some wish will have the same id as one of developer teams.

Reply:
This is why we also know the type - we will search for comments for item with given ID and of given type (e.g. wish type).

Reply:
If int is our choice, then we definitely need a type field in our Comment Type.

Awaiting for your response please

Reply:
> ID: The ID scalar type represents a unique identifier, often used to refetch an object or as the key for a cache. The ID type is serialized in the same way as a String; however, defining it as an ID signifies that it is not intended to be human‐readable.
https://graphql.org/learn/schema/

We can use ID type.

Reply:
We can use targetId as ID type + targetType as enum type of which one values will be 'wish'